src/mbgl/renderer/group_by_layout.cpp

```diff
@@ -18,7 +18,7 @@ std::string layoutKey(const RenderLayer& layer) {
     rapidjson::Writer<rapidjson::StringBuffer> writer(s);
 
     writer.StartArray();
-    writer.Uint(static_cast<uint32_t>(layer.type));
+    writer.Uint64(reinterpret_cast<uint64_t>(layer.baseImpl->getTypeInfo()));
```

Comment:
I think, now `friend std::string layoutKey(const RenderLayer&);` can be removed from `RenderLayer`.